### PR TITLE
Show number of Pokeballs in inventory when selecting a pokeball in the reward phase

### PR DESCRIPTION
**Pull request by happinyz**
QOL enhancement to show the user how many of the selected Pokeball they have while in the rewards phase.

image

### CODE REVIEW

File: src/modifier/modifier-type.ts [hunk 1]
```diff
@@ -145,8 +145,14 @@ class AddPokeballModifierType extends ModifierType implements Localizable {
 
   localize(): void {
     this.name = `${this.count}x ${getPokeballName(this.pokeballType)}`;
-    this.description = `Receive ${getPokeballName(this.pokeballType)} x${this.count}\nCatch Rate: ${getPokeballCatchMultiplier(this.pokeballType) > -1 ? `${getPokeballCatchMultiplier(this.pokeballType)}x` : 'Certain'}`;
+    this.description = `Receive ${getPokeballName(this.pokeballType)} x${this.count} (Inventory: {AMOUNT}) \nCatch Rate: ${getPokeballCatchMultiplier(this.pokeballType) > -1 ? `${getPokeballCatchMultiplier(this.pokeballType)}x` : 'Certain'}`;
   }
+  
+  getDescription(scene: BattleScene): string {
+    this.localize();
+    return this.description.replace('{AMOUNT}', scene.pokeballCounts[this.pokeballType].toLocaleString('en-US'));
```
Comment (TempsRay) [inline src/modifier/modifier-type.ts:153]: Does this need to be en-US? Isn't it just a number?

File: src/modifier/modifier-type.ts [hunk 2]
```diff
@@ -145,8 +145,14 @@ class AddPokeballModifierType extends ModifierType implements Localizable {
 
   localize(): void {
     this.name = `${this.count}x ${getPokeballName(this.pokeballType)}`;
-    this.description = `Receive ${getPokeballName(this.pokeballType)} x${this.count}\nCatch Rate: ${getPokeballCatchMultiplier(this.pokeballType) > -1 ? `${getPokeballCatchMultiplier(this.pokeballType)}x` : 'Certain'}`;
+    this.description = `Receive ${getPokeballName(this.pokeballType)} x${this.count} (Inventory: {AMOUNT}) \nCatch Rate: ${getPokeballCatchMultiplier(this.pokeballType) > -1 ? `${getPokeballCatchMultiplier(this.pokeballType)}x` : 'Certain'}`;
```
Comment (TempsRay) [inline src/modifier/modifier-type.ts:148]: Does Inventory need to be localized?

**Review by TempsRay — COMMENTED** [review]
LGTM besides some questions I had on localization. Good QOL update

File: src/modifier/modifier-type.ts (hunk 1 above)
Comment (happinyz) [reply src/modifier/modifier-type.ts]: Good point, changed to use `toString`

File: src/modifier/modifier-type.ts (hunk 2 above)
Comment (happinyz) [reply src/modifier/modifier-type.ts:149]: I'm not super familiar with i18n, but it seems like here we weren't doing any localization in the first place, despite the `localize` function existing. I agree we should be localizing this, but it's out of the scope of this PR. I added a comment to remind us to actually use i18n in the future.

**Review by TempsRay — APPROVED** [review]

**Review by Madmadness65 — APPROVED** [review]

**Pull request merged**